geth/node/rpc.go

```diff
@@ -94,6 +110,66 @@ func (c *RPCManager) Call(inputJSON string) string {
 	return c.makeJSONErrorResponse(ErrRPCServerTimeout)
 }
 
+// callNodeStream delivers giving request and body content to the external ethereum
+// (infura) RPCServer to process the request and returns response.
+func (c *RPCManager) callUpstreamStream(config *params.NodeConfig, body io.Reader) ([]byte, error) {
+	upstreamURL, err := url.Parse(config.UpstreamConfig.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq := httptest.NewRequest("POST", "/", body)
+	httpReq.RequestURI = ""
+	httpReq.URL = upstreamURL
+
+	httpClient := http.Client{
+		Timeout: 20 * time.Second,
+	}
+
+	res, err := httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	if respStatusCode := res.StatusCode; respStatusCode != http.StatusOK {
+		log.Error("handler returned wrong status code", "got", respStatusCode, "want", http.StatusOK)
+		return nil, ErrRPCServerCallFailed
+	}
+
+	defer res.Body.Close()
+
+	var resbody bytes.Buffer
+
+	if _, err := io.Copy(&resbody, res.Body); err != nil && err != io.EOF {
+		return nil, err
+	}
+
+	return resbody.Bytes(), nil
+}
+
+// callNodeStream delivers giving request and body content to the internal ethereum
+// RPCServer to process the request.
+func (c *RPCManager) callNodeStream(body io.Reader) ([]byte, error) {
+	server, err := c.nodeManager.RPCServer()
+	if err != nil {
+		// return c.makeJSONErrorResponse(err)
+		return nil, err
+	}
+
+	httpReq := httptest.NewRequest("POST", "/", body)
```

Comment:
Please see https://stackoverflow.com/questions/19595860/http-request-requesturi-field-when-making-request-in-go